ProcessLib/Process.h

```diff
@@ -200,6 +203,10 @@ class Process
 
     VectorMatrixAssembler _global_assembler;
 
+    /// Coupled processes.
+    mutable std::weak_ptr<StaggeredCouplingTerm> _coupling_term;
```

Comment:
IMHO it could be a `StaggeredCouplingTerm* _coupling_term`, and it probably does not have to be mutable:
* no weak_ptr, because it is unlikely that the time loop destroys processes while they are still running
* not mutable, because if you have
```
struct S { int* i; };
S const obj;
```
you could still do: `obj->i = 8`. I.e.: The value of `i` is mutable although `obj` is not. Cf. `int const*` vs `int* const`.